src/lib/update-lock.ts

```diff
@@ -72,6 +76,15 @@ async function dispose(
 	}
 }
 
+/**
+ * Composition step used by Supervisor compose module.
+ * Release all locks for an appId | appUuid.
+ */
+export async function releaseLock(appId: number) {
+	const release = await writeLock(appId);
```

Comment:
I think the `writeLock` function needs a better name. It took me a second to figure out this step was taking the process lock. Maybe `takeGlobalLockRW` and `takeGlobalLockRO` (instead of `readLock`)

Reply:
Makes sense!

Reply:
Updated names to be more clear [here](https://github.com/balena-os/balena-supervisor/pull/2234/commits/9d8241c9508c85012df309b275a5bdcef04b57bd#diff-d722e5c234dcc829cd6f5a4eb70cfdb9bf414361584d50bfa07fe228db711a41R22-R34)